api/wallet.go

```diff
@@ -0,0 +1,21 @@
+package api
+
+import (
+	"context"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/crypto"
+	"github.com/filecoin-project/lotus/chain/types"
+)
+
+type Wallet interface {
+	WalletNew(context.Context, types.KeyType) (address.Address, error) //perm:admin
+	WalletHas(context.Context, address.Address) (bool, error)          //perm:admin
+	WalletList(context.Context) ([]address.Address, error)             //perm:admin
+
+	WalletSign(ctx context.Context, signer address.Address, toSign []byte) (*crypto.Signature, error) //perm:admin
```

Comment:
The reason I am including `Wallet interface` here is:
1. LocalWallet in Lotus uses the following signature:
```
func (w *LocalWallet) WalletSign(ctx context.Context, addr address.Address, msg []byte, meta api.MsgMeta) (*crypto.Signature, error)
```

2. FullNode API uses a different signature:
```
	// WalletSign signs the given bytes using the given address.
	WalletSign(context.Context, address.Address, []byte) (*crypto.Signature, error) //perm:sign
```

Ideally I'd like for us to be able to use the `DealClient` both with a FullNode and with a Gateway (i.e. only local Wallet), hence the import and modification of a few dependencies.